Code/GraphMol/RascalMCES/RascalMCES.cpp

```diff
@@ -982,18 +985,54 @@ void calcDistMatrix(const std::vector<std::vector<int>> &adjMatrix,
   }
 }
 
+// Set the atomic number of the atoms that match the SMARTS in
+// RascalOptions.EquivalentAtoms to 110, 111 etc.  These will
+// be mapped back at the end.
+void assignEquivalentAtoms(ROMol &mol, const std::string &equivalentAtoms) {
+  if (equivalentAtoms.empty()) {
+    return;
+  }
+  std::vector<std::string> classSmarts;
+  boost::split(classSmarts, equivalentAtoms, boost::is_any_of(" "),
+               boost::token_compress_on);
+  if (classSmarts.size() > 9) {
```

Comment:
This seems like an arbitrary restriction, won't the code work with more than 9?

Reply:
As it says in the opening comment, that depends on what RDKit thinks of atomic numbers above 118. I assumed it would object though I didn’t check. Atoms that match a SMARTS class are all given the same atomic number starting at 110- it was a quick and dirty implementation. It could start lower than 110 but I was being exceptionally paranoid about what might be fed into it.

Reply:
I think it probably would still work with larger values, but (famous last words) "it's unlikely to ever really be a problem"

Reply:
Yes, I can’t think of a case where you’d want to have more than 9 types of equivalent atom. But, as you imply, you can never trust a chemist!